pkg/services/ngalert/prom/convert.go

```diff
@@ -18,18 +19,25 @@ const (
 	// alert rule when converting it to a Grafana alert rule. If this label is not present,
 	// a stable UID will be generated automatically based on the rule's data.
 	ruleUIDLabel = "__grafana_alert_rule_uid__"
-)
 
-const (
 	queryRefID          = "query"
 	prometheusMathRefID = "prometheus_math"
 	thresholdRefID      = "threshold"
 )
 
+var (
+	ErrInvalidDatasourceType = errutil.ValidationFailed("alerting.invalidDatasourceType")
+)
+
 // Config defines the configuration options for the Prometheus to Grafana rules converter.
 type Config struct {
+	// DataSourceUID is the UID of the datasource the rules are querying.
 	DatasourceUID  string
 	DatasourceType string
+	// TargetDatasourceUID is the UID of the datasource the recording rules are writing to.
+	// If not set, it defaults to DataSourceUID.
+	TargetDatasourceUID  string
+	TargetDatasourceType string
```

Comment:
`TargetDatasourceType` is used only to validate the type, but I decided to add it here for consistency because the `DatasourceType` is already here.